### PR TITLE
IAM 767: openfga dev setup

### PR DESCRIPTION
**Pull request by shipperizer**
IAM-767: openfga setup via skaffold

- **fix: enhance openfga client with CreateStore and helpers to set modelID and storeID on the fly**
- **fix: enhance cli model creation to bootstrap a store**
- **ci: swap openfga init job with admin ui create model one**
- **ci: drop init container and any trace of pv for openfga**
- **fix: allow for k8s client to be configured using kubeconfig**
- **fix: adapt serve command to changes on k8s client**
- **feat: allow create-fga-model cli command to save on a k8s coonfigmap**
- **ci: drop sudo cmd for skopeo copy**
- **ci: add openfga env vars to configmap**
- **docs: add little explanation on how to use new setup**

### CODE REVIEW

File: cmd/createFgaModel.go [hunk 1]
```diff
@@ -79,6 +90,49 @@ func createModel(apiUrl, apiToken, storeId string) {
...
+	k8sClient, err := k8s.NewCoreV1Client(kubeconfig)
+
+	if err != nil {
+		return err
+	}
+
+	cmNamespace := ""
+	cmName := ""
+
+	if k8sResource := strings.Split(configMap, "/"); len(k8sResource) == 2 {
```
Comment (wood-push-melon) [inline cmd/createFgaModel.go:113]: Nitpick(non-blocking):

Maybe this is better and consistent with other error handling?

```golang
parts := strings.Split(configMap, "/")
if len(parts) != 2 {
	return fmt.Errorf("invalid format for configmap resource %s: expected namespace/name", configMap)
}
namespace, name := parts[0], parts[1]
```

**Comment by wood-push-melon** [comment]
LGTM. Just one question: do you think we should put something into the `README.md` for how to do after people change the model? (e.g. restart the admin service from what I understand) for other teams to know.

**Comment by shipperizer** [comment]
> LGTM. Just one question: do you think we should put something into the `README.md` for how to do after people change the model? (e.g. restart the admin service from what I understand) for other teams to know.

have another go at the README.md, lemme know if that s enough

File: README.md [hunk 2]
```diff
@@ -56,6 +61,14 @@ As a requirement, please make sure to:
 
 Run `make dev` to get a working environment in k8s
 
+### OpenFGA initialization
+
+The Admin Service comes up with authorization disabled (see `AUTHORIZATION_ENABLED` env var), The env vars `OPENFGA_AUTHORIZATION_MODEL_ID` and `OPENFGA_STORE_ID` which are needed for the correct functioning of the RBAC APIs get set by the job `admin-ui-openfga-setup`, after this has completed a developer is supposed to bounce the deployment to get the application to source the new env vars, setting `AUTHORIZATION_ENABLED` will make sure those endpoints use OpenFGA as a backend instead of a `NoOp implementation` (behaviour will change,  see https://github.com/canonical/identity-platform-admin-ui/issues/221)
```
Comment (nsklikas) [inline README.md:66]: To make it clearer, we could add the commands that the dev will have to run, I suppose it's something like:
```console
# Wait for the openfga setup job to complete
kubectl wait --for=condition=complete job/admin-ui-openfga-setup

# Edit the configmap to enable authorization by setting AUTHORIZATION_ENABLED=true
kubectl edit configmap identity-platform-admin-ui

# Restart the admin UI apply the changes
kubectl rollout restart deployment identity-platform-admin-ui
```

File: README.md [hunk 3]
```diff
@@ -56,6 +61,14 @@ As a requirement, please make sure to:
...
+
+As k8s jobs on their own don't get deleted, make sure that to run `kubectl delete job admin-ui-openfga-setup` if you are planning to change the model, make sure environment variables in the `identity-platform-admin-ui` configmap reflect the status you want
```
Comment (nsklikas) [inline README.md:68]: Again to make it clearer we should provide the steps that are needed in order to change the model. Something like

```
If you wish to make changes to the openfga model, you need to make sure that the job for setting up openfga is deleted before redeploying the the admin UI:

kubectl delete job admin-ui-openfga-setup
make dev
```

File: deployments/kubectl/configMap.yaml [hunk 4]
```diff
@@ -18,6 +18,9 @@ data:
   RULES_CONFIGMAP_NAME: oathkeeper-rules
   RULES_CONFIGMAP_FILE_NAME: access-rules.json
   RULES_CONFIGMAP_NAMESPACE: default
+  OPENFGA_API_SCHEME: http
+  OPENFGA_API_HOST: openfga.default.svc.cluster.local:8080
+  OPENFGA_API_TOKEN: "42"
```
Comment (nsklikas) [inline deployments/kubectl/configMap.yaml:23]: Should we add: 
```
AUTHORIZATION_ENABLED: "false"
```

To make editing the configmap easier after openfga setup is complete?

**Review by nsklikas — APPROVED** [review]
LGTM. Left some comments for improving the readme.

**Review by BarcoMasile — APPROVED** [review]

**Review by nsklikas — APPROVED** [review]

**Review by wood-push-melon — APPROVED** [review]

**Pull request merged**